### PR TITLE
NodeJS: fix incorrect dependency on through2

### PR DESCRIPTION
**Pull request by landrito**
Through2 should not be used if the api does not have a streaming method.

Fixes: #1585

### CODE REVIEW

**Comment by JustinBeckwith** [comment]
@alexander-fenster @lukesneeringer this has been sitting around for quite a while :) Can we merge it?

File: src/main/java/com/google/api/codegen/transformer/nodejs/NodeJSPackageMetadataTransformer.java [hunk 1]
```diff
@@ -215,34 +216,34 @@ private ViewModel generateMetadataView(
...
 
-  private boolean hasLongrunning(ApiModel model, ProductConfig productConfig) {
-    for (InterfaceModel apiInterface : model.getInterfaces()) {
-      if (productConfig.getInterfaceConfig(apiInterface).hasLongRunningOperations()) {
-        return true;
-      }
+  private List<PackageDependencyView> generateDevDependencies(
+      ApiModel model, ProductConfig productConfig) {
+    ImmutableList.Builder<PackageDependencyView> dependencies = ImmutableList.builder();
+    dependencies.add(PackageDependencyView.create("mocha", VersionBound.create("3.2.0", "")));
```
Comment (alexander-fenster) [inline src/main/java/com/google/api/codegen/transformer/nodejs/NodeJSPackageMetadataTransformer.java]: hey, I hate this: this feels very wrong to have version numbers hardwired in the code, even though it's Node-specific class.

**Comment by alexander-fenster** [comment]
I don't really know the code and I don't want to sound picky but I'd rather leave the `package.json` content and especially versions of the dependencies to snip files. I might as well be very wrong here so feel free to approve and merge but I'd like someone to express the second opinion :)  @lukesneeringer, @eoogbe?

**Comment by eoogbe** [comment]
I would prefer to move as much of the hard coded stuff as possible to the snippets. In fact, that is going to be a part of the view simplification refactoring for Ruby. That said, the current standard is to keep as much stuff as possible out of the snippets and inside the transformers. Your call if you want to keep things more consistent with the rest of the codebase or change to my preference.

**Comment by garrettjonesgoogle** [comment]
Potentially dependencies seem like a good thing to keep out of the snippets, because they are a point of frequent updates and they probably need to be updated together.

**Pull request closed without merging**